src/WorldStorage/FastNBT.cpp

```diff
@@ -91,9 +91,9 @@ bool cParsedNBT::ReadString(size_t & a_StringStart, size_t & a_StringLen)
 	NEEDBYTES(2);
 	a_StringStart = m_Pos + 2;
 	a_StringLen = static_cast<size_t>(GetBEShort(m_Data + m_Pos));
-	if (a_StringLen > 0xffff)
+	if (a_StringLen > m_Length - m_Pos - 2)
```

Comment:
No it can't without UB. If `m_Pos > m_Length` then we hit UB at line 93.
And if `2 < m_Length - m_Pos` then we should of exited in NEEDBYTES.